test/gtest/common/test.cc

```diff
@@ -94,10 +94,12 @@ void test_base::get_config(const std::string& name, std::string& value, size_t m
 }
 
 void test_base::modify_config(const std::string& name, const std::string& value,
-                              bool optional)
+                              bool optional, bool found)
 {
     ucs_status_t status = ucs_global_opts_set_value(name.c_str(), value.c_str());
-    if ((status == UCS_ERR_NO_ELEM) && optional) {
+    if ((status != UCS_OK) && found) {
+        return;
+    } else if ((status == UCS_ERR_NO_ELEM) && optional) {
```

Comment:
why need to add "found"? how is it different than the parameter "optional"?

Reply:
'found' is needed only from uct's modify_config. 'optional' is passed to this function from other places and from uct' modify_config as well. 
do you want uct's modify_config to override the value of 'optional'?

Reply:
yes, it can set "optional" parameter to "true", to avoid the call to GTEST_FAIL() below